packages/react/src/components/Notice/index.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { forwardRef } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import Icon, { iconTypes, IconType } from '../Icon';
+
+const iconTypeMap = {
+  caution: 'caution',
+  info: 'info-circle'
+};
+
+export interface NoticeProps
+  extends Omit<React.HTMLAttributes<HTMLDivElement>, 'title'> {
+  type?: keyof typeof iconTypeMap;
+  title?: React.ReactNode;
+  icon?: IconType;
+  children?: React.ReactNode;
+}
+
+/**
+ * The cauldron Notice notification component
+ */
+const Notice = forwardRef<HTMLDivElement, NoticeProps>(
+  ({ type, title, icon, children, ...otherProps }: NoticeProps, ref) => {
+    const validType =
+      type && Object.keys(iconTypeMap).includes(type) ? type : 'info';
+    const validIcon =
+      icon && iconTypes.includes(icon)
+        ? icon
+        : (iconTypeMap[validType] as IconType);
```

Comment:
Alternatively, we could skip the lookup completely here for `iconTypes` as we could rely on typescript to do the validation for us. What do you think?

Reply:
I'm fine with either one, although without narrowing the value in some way we likely would need to either assert the type or allow for a generic. 

I'm hesitant to do that since if we make the interface overly generic, TS may not be as helpful as it typically would be.